tests/cpp_validation_tests/test_validation.cpp

```diff
@@ -40,238 +86,263 @@ auto read_json(std::filesystem::path const& path) {
     return j;
 }
 
-class UnsupportedValidationCase : public PowerGridError {
-  public:
-    UnsupportedValidationCase(std::string const& calculation_type, bool sym) {
-        using namespace std::string_literals;
-
-        auto const sym_str = sym ? "sym"s : "asym"s;
-        append_msg("Unsupported validation case: "s + sym_str + " "s + calculation_type);
-    };
-};
-
-// memory buffer
-using BufferPtr = std::unique_ptr<void, std::add_pointer_t<void(RawDataConstPtr)>>; // custom deleter at runtime
-struct Buffer {
-    BufferPtr ptr{nullptr, [](void const*) {}};
-    IdxVector indptr;
+struct OwningMemory {
+    std::vector<Buffer> buffers;
+    std::vector<std::vector<Idx>> indptrs;
 };
 
 struct OwningDataset {
-    MutableDataset dataset;
-    ConstDataset const_dataset;
-    std::vector<Buffer> buffers{};
-    std::vector<ConstDataset> batch_scenarios{};
+    std::optional<DatasetMutable> dataset;
+    std::optional<DatasetConst> const_dataset;
+    OwningMemory storage{};
 };
 
-auto create_owning_dataset(WritableDataset& info) {
+OwningDataset create_owning_dataset(DatasetWritable& writable_dataset) {
+    auto const& info = writable_dataset.get_info();
+    Idx const is_batch = info.is_batch();
     Idx const batch_size = info.batch_size();
-    std::vector<Buffer> buffers;
+    auto const& dataset_name = info.name();
+    OwningDataset owning_dataset{.dataset{DatasetMutable{dataset_name, is_batch, batch_size}},
+                                 .const_dataset = std::nullopt};
 
     for (Idx component_idx{}; component_idx < info.n_components(); ++component_idx) {
-        auto const& component_info = info.get_component_info(component_idx);
-        auto const& component_meta = component_info.component;
-
-        Buffer buffer{};
-        buffer.ptr =
-            BufferPtr{component_meta->create_buffer(component_info.total_elements), component_meta->destroy_buffer};
-        buffer.indptr = IdxVector(component_info.elements_per_scenario < 0 ? batch_size + 1 : 0);
-        Idx* indptr_data = buffer.indptr.empty() ? nullptr : buffer.indptr.data();
-
-        info.set_buffer(component_info.component->name, indptr_data, buffer.ptr.get());
-        buffers.push_back(std::move(buffer));
+        auto const& component_name = info.component_name(component_idx);
+        auto const& component_meta = MetaData::get_component_by_name(dataset_name, component_name);
+        Idx const component_elements_per_scenario = info.component_elements_per_scenario(component_idx);
+        Idx const component_size = info.component_total_elements(component_idx);
+
+        auto& current_intptr = owning_dataset.storage.indptrs.emplace_back(
```

Comment:
indptr for index pointer, not for integer pointer 😬 

```suggestion
        auto& current_indptr = owning_dataset.storage.indptrs.emplace_back(
```

Reply:
Oops. Thanks for catching.